compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -59,6 +59,7 @@ class Compiler {
   protected def transformPhases: List[List[Phase]] =
     List(new FirstTransform,         // Some transformations to put trees into a canonical form
          new CheckReentrant,         // Internal use only: Check that compiled program has no data races involving global vars
+         new CheckExperimental,      // Check that no @experimental APIs are used
```

Comment:
I suggest doing this check before pickling, so we don't emit tasty which uses experimental features. Maybe as part of PostTyper?

Reply:
Indeed, my plan is to move it as early as possible. In this first iteration I'm trying to figure out what needs to be checked and test cases. Note that I already moved checking of inline definition to typer. Post typer seems to be the logical place for the rest of the checks.

Reply:
OK, I think even the inlined calls could be checked in PostTyper since we retain Inlined#call

Reply:
That might not work if we have a nested inline that removed the other inline code.

Reply:
Moved checks to PostTyper